### PR TITLE
Expose decorationThickness in TextStyle. Roll engine (12 commits)

### PR DESCRIPTION
**Pull request by GaryQian**
Fixes part of https://github.com/flutter/flutter/issues/28704

This exposes decorationThickness to painting TextStyle, which is an already implemented but hidden property in LibTxt. 

Tests verify dart:ui receives the decoration thickness. The LibTxt tests verify that the thickness is respected.

git log --oneline --no-merges 5ccee95373348854bc4877cfe240024a36847d2d..31b289f277c653ea91e20d75309958ac0dd0b865
31b289f27 Fix indexing error in dart:ui TextStyle.toString (flutter/engine#8143)
fc2e6b61c Typo "fast an inline" to "fast and inline" (flutter/engine#8142)
0f19b2de0 Reland PerformanceOverlayLayer golden test (flutter/engine#8140)
073aadde1 Fix TextStyle decode misalignment (flutter/engine#8141)
d87d29054 Roll src/third_party/skia 406b068942f0..2eecc3ea3d71 (11 commits) (flutter/engine#8138)
5cef4a022 Use final state passed to dart before initialization as the initial lifecycleState. (flutter/engine#8124)
ffef51be6 Roll src/third_party/skia 665bc64a2dc4..406b068942f0 (8 commits) (flutter/engine#8137)
48efd0fb7 Roll src/third_party/skia 762ddd7e4352..665bc64a2dc4 (2 commits) (flutter/engine#8129)
f666adbbd Roll src/third_party/skia 2932a458957d..762ddd7e4352 (3 commits) (flutter/engine#8128)
8b0df6ded Bugfix #29203: NPE in getAccessibilityProvider in old FlutterView. (flutter/engine#8126)
8f7b18345 Roll src/third_party/skia c6d8781c4036..2932a458957d (2 commits) (flutter/engine#8125)
52b67fdd5 Expose decorationThickness to dart:ui (flutter/engine#8008)

### CODE REVIEW

File: packages/flutter/lib/src/painting/text_style.dart [hunk 1]
```diff
@@ -490,6 +491,10 @@ class TextStyle extends Diagnosticable {
   /// The style in which to paint the text decorations (e.g., dashed).
   final TextDecorationStyle decorationStyle;
 
+  /// The thickness of the decoration as a muliplier of the thickness specified
+  /// by the font.
```
Comment (Hixie) [inline packages/flutter/lib/src/painting/text_style.dart]: this needs to be more detailed. What does "thickness" mean? What is the default? Maybe a screenshot would explain it better? (see the assets-for-api-docs repo for how to make diagrams)

**Comment by GaryQian** [comment]
Will require an engine roll with https://github.com/flutter/engine/pull/8008

**Comment by GaryQian** [comment]
(for reference) Have spoken with Hixie and have approval on the addition of the new `decorationThickness` property from an API design standpoint.

File: packages/flutter/lib/src/painting/text_style.dart [hunk 2]
```diff
@@ -490,6 +491,19 @@ class TextStyle extends Diagnosticable {
   /// The style in which to paint the text decorations (e.g., dashed).
   final TextDecorationStyle decorationStyle;
 
+  /// The thickness of the decoration stroke as a muliplier of the thickness
+  /// defined by the font.
+  ///
+  /// The font provides a base stroke width for decorations which scales off
```
Comment (goderbauer) [inline packages/flutter/lib/src/painting/text_style.dart]: This doc would probably be slightly more accessible if it gave one or two concrete examples of what a decoration stroke may be that's affected by this property.

File: packages/flutter/test/material/theme_test.dart [hunk 3]
```diff
@@ -721,7 +721,7 @@ class _TextStyleProxy implements TextStyle {
   }
 
   @override
-  TextStyle apply({ Color color, Color backgroundColor, TextDecoration decoration, Color decorationColor, TextDecorationStyle decorationStyle, String fontFamily, List<String> fontFamilyFallback, double fontSizeFactor = 1.0, double fontSizeDelta = 0.0, int fontWeightDelta = 0, double letterSpacingFactor = 1.0, double letterSpacingDelta = 0.0, double wordSpacingFactor = 1.0, double wordSpacingDelta = 0.0, double heightFactor = 1.0, double heightDelta = 0.0 }) {
+  TextStyle apply({ Color color, Color backgroundColor, TextDecoration decoration, Color decorationColor, TextDecorationStyle decorationStyle, double decorationThicknessFactor = 1.0, double decorationThicknessDelta = 0.0, String fontFamily, List<String> fontFamilyFallback, double fontSizeFactor = 1.0, double fontSizeDelta = 0.0, int fontWeightDelta = 0, double letterSpacingFactor = 1.0, double letterSpacingDelta = 0.0, double wordSpacingFactor = 1.0, double wordSpacingDelta = 0.0, double heightFactor = 1.0, double heightDelta = 0.0 }) {
```
Comment (goderbauer) [inline packages/flutter/test/material/theme_test.dart]: nit: this and the next change lines would be more readable with line breaks (e.g. one argument per line).

**Review by goderbauer — APPROVED** [review]
LGTM with two nits.

File: packages/flutter/test/material/theme_test.dart (hunk 3 above)
Comment (GaryQian) [reply packages/flutter/test/material/theme_test.dart]: I'll go ahead and change the other super-long lines to multi-lines also.

**Comment by GaryQian** [comment]
Mistake fixed in https://github.com/flutter/engine/pull/8141, will need to be rolled to it in order to pass.

**Pull request merged**